srai/loaders/osm_way_loader/osm_way_loader.py

```diff
@@ -0,0 +1,373 @@
+            contain_within_area (bool): defaults to False
+                Whether to remove the roads that have one of their nodes outside of the given area.
+            preprocess (bool): defaults to True
+                Whether to preprocess the data.
+            wide (bool): defaults to True
+                Whether to return the roads in wide format.
+            osm_way_tags (List[str]): defaults to constants.OSM_WAY_TAGS
+                Dict of tags to take into consideration during computing.
+        """
+        import_optional_dependencies(dependency_group="osm", modules=["osmnx"])
+
+        self.network_type = network_type
+        self.contain_within_area = contain_within_area
+        self.preprocess = preprocess
+        self.wide = wide
+        self.osm_keys = list(osm_way_tags.keys())
+        self.osm_tags_flat = (
+            seq(osm_way_tags.items())
+            .flat_map(lambda x: [f"{x[0]}-{v}" if x[0] not in ("oneway") else x[0] for v in x[1]])
+            .distinct()
+            .to_list()
+        )
+
+    def load(self, area: gpd.GeoDataFrame) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+        """
+        Load road infrastructure for a given GeoDataFrame.
+
+        Args:
+            area (gpd.GeoDataFrame): (Multi)Polygons for which to download road infrastructure data.
+
+        Raises:
+            ValueError: If provided GeoDataFrame has no crs defined.
+            ValueError: If provided GeoDataFrame is empty.
+            TypeError: If provided geometries are not of type Polygon or MultiPolygon.
+            LoadedDataIsEmptyException: If all of the supplied area polygons do not contain
+                any road infrastructure data.
+
+        Returns:
+            Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]: Road infrastructure as (intersections, roads)
+        """
+        import osmnx as ox
+
+        ox.settings.useful_tags_way = constants.OSMNX_WAY_KEYS
+        ox.settings.timeout = constants.OSMNX_TIMEOUT
+
+        if area.empty:
+            raise ValueError("Provided `area` GeoDataFrame is empty.")
+
+        gdf_wgs84 = area.to_crs(crs=WGS84_CRS)
+
+        gdf_nodes_raw, gdf_edges_raw = self._graph_from_gdf(gdf_wgs84)
+        if gdf_edges_raw.empty or gdf_edges_raw.empty:
+            raise LoadedDataIsEmptyException(
+                "It can happen when there is no road infrastructure in the given area."
+            )
+
+        gdf_edges = self._explode_cols(gdf_edges_raw)
+
+        if self.preprocess:
+            self._preprocess(gdf_edges, inplace=True)
+
+        if self.wide:
+            gdf_edges = self._to_wide(gdf_edges_raw, gdf_edges)
+
+        return gdf_nodes_raw, gdf_edges
+
+    def _graph_from_gdf(self, gdf: gpd.GeoDataFrame) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+        """
+        Obtain the raw road infrastructure data from OSM.
+
+        Args:
+            gss (gpd.GeoSeries): (Multi)Polygons for which to download road infrastructure data.
```

Comment:
This is different from what is said in the function's signature. Is it a GeoDataFrame or a GeoSeries?